### ISSUE TITLE
Piano keyboard input type requests

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap#2897]
I have two (hopefully minor) requests for the piano keyboard input type:

1. Can the input slot please be one you can type into in addition to selecting a note from the piano? The piano keyboard selection feature is fantastic, but it's slower than typing, and it would be nice to have both options available  for inputting notes.
2. Can that input type please allow for multiple inputs so you have little triangles on the block (like join and list do)? It would be nice to be able to add additional notes without having to edit the block and so you don't have to make multiple blocks with different numbers of note inputs.

If these changes are quick (I'm optimistic, I know), it would be fantastic to have them in time for developing the next unit of BJC Middle School in November. Thanks!

**Comment by brianharvey** [comment]
I think this request is based on a confusion about how inputs work.  The input _type_ (e.g., piano key) is orthogonal to the variadicness.  If you want a variadic piano key input, then that's what you should ask for in the long-form input dialog.

The author of a custom block has to know whether or not the input is variadic, because if so, it's really an input of type List as far as the block definition is concerned.

Rreally what you're asking for isn't a convenience, but rather a _restriction_ on users of this input type; they can't write the simple block they want to write (for example, to turn a piano key into a MIDI number) but have to deal with lists instead.

If what you really have in mind is that some _primitive_ sound block should have a variadic piano-key input, say that, and say which primitive.

**Comment by cycomachead** [comment]
> I think this request is based on a confusion about how inputs work. The input _type_ (e.g., piano key) is orthogonal to the variadicness. If you want a variadic piano key input, then that's what you should ask for in the long-form input dialog.

That is not true in the way Snap! actually _works_ even though it should be. Not all inputs can be variadic. 
Really I think the issue is that _dropdown_ inputs cannot be variadic, of which the piano is a type.

I think the request is that a user should be able to represent some list of notes inputs but be able to do so with the piano keyboard option.

**Comment by ego-lay-atman-bay** [comment]
something like this?
![image](https://user-images.githubusercontent.com/70973578/141406890-840fed2c-0a81-4d56-aef1-bbe720a07ca8.png)
this was made by editing the xml, so all you would need to do is add it to the ui.

**Comment by maryfries** [comment]
Yes! Exactly! Are there supposed to be two sets of left/right arrows though? Thanks! :)

**Comment by ego-lay-atman-bay** [comment]
No, that's just my block.
